python/dataset.cpp

```diff
@@ -82,15 +94,32 @@ void bind_mutable_view(py::module &m, const std::string &name) {
       .def(
           "keys", [](T &self) { return keys_view(self); },
           py::keep_alive<0, 1>(), R"(view on self's keys)")
-      .def(
-          "values", [](T &self) { return values_view(self); },
-          py::keep_alive<0, 1>(), R"(view on self's values)")
       .def(
           "items", [](T &self) { return items_view(self); },
           py::return_value_policy::move, py::keep_alive<0, 1>(),
-          R"(view on self's items)")
-      .def("__contains__", &T::contains);
-  bind_inequality_to_operator<T>(view);
+          R"(view on self's items)");
+}
+
+template <class T, class ConstT>
+void bind_mutable_view_no_dim(py::module &m, const std::string &name) {
+  py::class_<ConstT>(m, (name + "ConstView").c_str());
+  py::class_<T, ConstT> view(m, (name + "View").c_str());
+  bind_common_mutable_view_operators<T>(view);
+  view.def(
+          "__iter__",
+          [](T &self) {
+            auto keys_view = str_keys_view(self);
+            return py::make_iterator(keys_view.begin(), keys_view.end(),
+                                     py::return_value_policy::move);
+          },
+          py::keep_alive<0, 1>())
+      .def(
+          "keys", [](T &self) { return str_keys_view(self); },
```

Comment:
Is `str_keys_view` really necessary? Can we just return a vector of the (string) keys and get essentially the same behavior?

Reply:
Potentially we could. My only worry would be that this may lead to the `keys` method in python returning a list object rather than the iterable it returns as standard in python3. This was why I assumed the keys_view was introduced in the first place but don't know how big of an issue it is.